packages/components/src/internal/components/lineage/models.ts

```diff
@@ -974,36 +967,220 @@ export function isAliquotNodeCollection(node: LineageNodeCollection | LineageNod
     return node.materialLineageType === 'Aliquot';
 }
 
-function _processNodes(
+function processCombinedNode(
+    lsid: string,
+    edge: LineageLink,
+    combinedNode,
+    visEdges: EdgeMap,
+    visNodes: VisNodeMap,
+    nodesInCombinedNode: { [key: string]: string[] }
+): void {
+    if (!combinedNode) return;
+
+    // remove the basic node from the graph if it exists
+    if (visNodes[edge.lsid]) {
+        if (visNodes[edge.lsid].kind !== 'node') {
+            console.error('Edge in raw graph should only connect basic nodes: ' + lsid + ' -> ' + edge.lsid);
+        }
+        delete visNodes[edge.lsid];
+    }
+
+    // Find existing edges in either direction to the edge target node being added to the combined node.
+    // The existing edges will be deleted and re-added to the new combined node.
+    const existingEdges = findConnectedNodes(Object.values(visEdges), edge.lsid);
+    // console.log("  ".repeat(depth) + "existing edges for '" + e.lsid + "': " + existingEdges.map(e => e.id));
```

Comment:
I assume these commented out console.log calls were left in intentionally. Just checking.

Reply:
I'd left them but this is good a time as any to remove them.

Reply:
Removed.